src/libsyntax/parse/parser.rs

```diff
@@ -6007,7 +6007,9 @@ impl<'a> Parser<'a> {
         }
 
         if self.check_keyword(keywords::Const) {
-            return Err(self.span_fatal(self.span, "extern items cannot be `const`"));
+            let mut err = self.span_fatal(self.span, "extern items cannot be `const`");
+            err.help("use `static` instead");
+            return Err(err);
```

Comment:
This can use better recovery - you can report a non-fatal error and then return the same successful result as `self.check_keyword(keywords::Static)` does.
You'll have to move `self.expect_keyword(keywords::Static)` from `parse_item_foreign_static` though.